src/averageAges.js

```diff
@@ -55,7 +78,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const motherAndSon = people.map((child) => {
+    return [child, people.find(
+      (mother) => onlyWithSon
+        ? mother.name === child.mother && child.sex === 'm'
+        : mother.name === child.mother)];
+  });
+
+  const filteredMotherAndSonAges = motherAndSon.filter((person) => {
+    return !person.includes(undefined);
```

Comment:
It looks a bit complicated. You can just filter to leave only children with mother (and only m if required).
And then using map method transform it into an array with age differences (you can find mother of child inside map method).